src/main/kotlin/org/opensearch/commons/notifications/NotificationsPluginInterface.kt

```diff
@@ -217,6 +221,30 @@ object NotificationsPluginInterface {
         )
     }
 
+    /**
+     * Publishes a notification API using the legacy notification implementation. No REST API.
+     * Internal API only for the Index Management plugin.
+     * @param client Node client for making transport action
+     * @param request The legacy publish notification request
+     * @param listener The listener for getting response
+     */
+    fun publishLegacyNotification(
+        client: NodeClient,
+        request: LegacyPublishNotificationRequest,
+        listener: ActionListener<LegacyPublishNotificationResponse>
+    ) {
+        if (request.feature != Feature.INDEX_MANAGEMENT) {
+            // Do not change this; do not pass in Feature.INDEX_MANAGEMENT if you are not the Index Management plugin.
+            throw IllegalArgumentException("The publish notification method only supports the Index Management feature.")
+        }
+
+        client.execute(
+            LEGACY_PUBLISH_NOTIFICATION_ACTION_TYPE,
+            request,
+            wrapActionListener(listener) { response -> recreateObject(response) { LegacyPublishNotificationResponse(it) } }
```

Comment:
Please let me know if recreateObject is required here. we may have to update in other transport calls as well. (as of now it seems it works without it)

Reply:
If I remember previously, this was required for class loader issue. All of the transport calls are following this way as defined in rest of fail.